rust/src/state_machine/mod.rs

```diff
@@ -1,3 +1,109 @@
+//! messages has been submitted and initializes the [`SeedDict`].
+//!
+//! **Update**
+//!
+//! Publishes the [`PhaseEvent::Update`], publishes the `scalar`, builds and publishes the
+//! [`SeedDict`], ensures that enough update messages has been submitted and aggregates the
+//! masked model.
+//!
+//! **Sum2**
+//!
+//! Publishes the [`PhaseEvent::Sum2`], builds the [`MaskDict`] and determines the applicable mask
+//! for unmasking the global masked model.
+//!
+//! **Unmask**
+//!
+//! Publishes the [`PhaseEvent::Unmask`], unmasks the global masked model and publishes the global
+//! model.
+//!
+//! **Error**
+//!
+//! Publishes the [`PhaseEvent::Error`] and handles [`StateError`]s that can occur during the
+//! execution of the [`StateMachine`]. In most cases, the error is handled by restarting the round.
+//! However, if a [`StateError::ChannelError`] occurs, the [`StateMachine`] will shut down.
+//!
+//! **Shutdown**
+//!
+//! Publishes the [`PhaseEvent::Shutdown`] and shuts down the [`StateMachine`]. During the shutdown,
+//! the [`StateMachine`] performs a clean shutdown of the [Request][requests_idx] channel by
+//! closing it and consuming all remaining messages.
+//!
+//! # Requests
+//!
+//! By initiating a new [`StateMachine`] via [`StateMachine::new()`], a new
+//! [Request][requests_idx] channel is created, the function of which is to send [`Request`]s to
+//! the [`StateMachine`]. The sender half of that channel ([`RequestSender`]) is returned back to
+//! the caller of [`StateMachine::new()`], whereas the receiver half ([`RequestReceiver`]) is used
+//! by the [`StateMachine`].
+//!
+//! <div class="information">
+//!     <div class="tooltip ignore" style="">ⓘ<span class="tooltiptext">Note</span></div>
+//! </div>
+//! <div class="example-wrap" style="display:inline-block">
+//! <pre class="ignore" style="white-space:normal;font:inherit;">
+//!     <strong>Note</strong>: <code>Requests</code> are only processed in the states
+//!     <code>Sum</code>, <code>Update</code> or <code>Sum2</code>.<br><br>
+//!     If the <code>Request</code> type and the state of the state machine do not match,
+//!     the <code>Request</code>  is ignored and the sender of the request receives a
+//!     <code>PetError::InvalidMessage</code>
+//! </pre></div>
+//!
+//! See [here][requests] for more details.
+//!
+//! # Events
+//!
+//! During the execution of the PET protocol, the [`StateMachine`] will publish various events
+//! (see Phase states). Everyone who is interested in the events can subscribe to the respective
+//! events via the [`EventSubscriber`]. A [`EventSubscriber`] is automatically created when a new
+//! [`StateMachine`] is created through [`StateMachine::new()`].
+//!
+//! See [here][events] for more details.
+//!
+//!
+//! [settings]: ../settings/index.html
+//! [`PhaseEvent::Idle`]: crate::state_machine::events::PhaseEvent::Idle
```

Comment:
it's good to know that one can also write links in the `use` syntax instead of html links, makes things easier :+1: